lighthouse-core/gather/driver.js

```diff
@@ -506,12 +557,19 @@ class Driver {
       options: 'sampling-frequency=10000'  // 1000 is default and too slow.
     };
 
-    // Disable any domains that could interfere or add overhead to the trace
-    return this.sendCommand('Debugger.disable')
-      .then(_ => this.sendCommand('CSS.disable'))
-      .then(_ => this.sendCommand('DOM.disable'))
-      // Enable Page domain to wait for Page.loadEventFired
-      .then(_ => this.sendCommand('Page.enable'))
+    // Check any domains that could interfere with or add overhead to the trace.
+    if (this.domainEnabled('Debugger')) {
```

Comment:
> If we're no longer disabling these before a trace, won't that change perf results?

In the current config these domains aren't enabled when we take a trace, so there's no change currently. If we add a gatherer that does enable one of these domains before a trace, though, it could cause a difference, which is why I kept a warning message. Maybe that's not enough, though.

> the log error message will propagate all the way into the report right? If it were an error, the DBW audit/gatherer would surface a -1 in the report?

`log.error` only logs it in red to the CLI/console, which hopefully someone catches :)

The problem with the approach on master is that it introduces errors but in the opposite way: it undercuts any gatherers that may unknowingly be running in the same pass as a trace being taken. The result may be subtle and not an error. e.g. a gather is listening for a `CSS` event that never comes because driver has disabled the `CSS` domain out from under it, but the lack of an event may just mean whatever it was gathering didn't happen on this particular page.

Maybe as suggested we do want to assert the domains are disabled and throw if they aren't, with a nice error message saying that the current config has gatherers interfering with a trace